### ISSUE TITLE
Compilation error with MonoAndroid 13 since with Visual Studio 2022 17.5 due to a Japanese character in a method name

### ISSUE THREAD

**Issue by vilinet** [martinjw/Holiday#137]
Visual Studio reworked some utf8 encoding quite a while ago and since it made this project compilation to fail when running the AOT compiler with monoandroid13 xamarin target.

The problematic character is the `ō` in the JapanPublicHoliday.cs: ShōwaDay(int year) method,
that character cannot be present for some reason in any identifier. (but it can be still a comment)

This issue is persistent for more than a year now with all the latest tool installed all the time. I have VS 17.11.5 at the moment installed.

The error produced is the following one:

`
[aot-compiler stdout] JIT time: 10 ms, Generation time: 14 ms, Assembly+Link time: 553 ms.
[aot-compiler stderr] obj\\DEV-Release\\130\\aot\\arm64-v8a\\PublicHoliday.dll\\temp.s:90931:48: error: unexpected token in directive
[aot-compiler stderr]         .local plt_PublicHoliday_JapanPublicHoliday_Sh┼_waDay_int
[aot-compiler stderr] obj\\DEV-Release\\130\\aot\\arm64-v8a\\PublicHoliday.dll\\temp.s:90932:47: error: expected STT_<TYPE_IN_UPPER_CASE>, '#<type>', '@<type>', '%<type>' or "<type>"
[aot-compiler stderr]         .type plt_PublicHoliday_JapanPublicHoliday_Sh┼_waDay_int,@function
[aot-compiler stderr] obj\\DEV-Release\\130\\aot\\arm64-v8a\\PublicHoliday.dll\\temp.s:90933:40: error: invalid operand
[aot-compiler stderr] plt_PublicHoliday_JapanPublicHoliday_Sh┼_waDay_int:
[aot-compiler stderr] obj\\DEV-Release\\130\\aot\\arm64-v8a\\PublicHoliday.dll\\temp.s:90940:47: error: unexpected token in directive
[aot-compiler stderr]         .size plt_PublicHoliday_JapanPublicHoliday_Sh┼_waDay_int,.-plt_PublicHoliday_JapanPublicHoliday_Sh┼_waDay_int
`

**Comment by wollac11** [comment]
Yeah this issue effectively means that with modern versions of MSBuild any app using the mono AOT compiler that references this library will not build. A bit weird since it worked with older versions but since MS seems to have no interest in fixing it, it would be better to avoid supported characters in class and member names.

**Comment by wollac11** [comment]
This issue is now fixed after PR#138 was merged (https://github.com/martinjw/Holiday/pull/138)
